core/services/vrf/listener_v2.go

```diff
@@ -571,12 +571,7 @@ func (lsn *listenerV2) processRequestsPerSubBatch(
 		}
 
 		fromAddresses := lsn.fromAddresses()
-		fromAddress, err := lsn.gethks.GetRoundRobinAddress(lsn.chainID, fromAddresses...)
-		if err != nil {
-			l.Errorw("Couldn't get next from address", "err", err)
-			continue
-		}
-		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddress)
+		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddresses[0])
```

Comment:
Should we do a empty slice check here on `fromAddresses` prior to accessing `fromAddresses[0]`?

Reply:
I think we should because we don't actually validate that `fromAddresses` is passed into the job spec. Maybe:

```suggestion
		if len(fromAddresses) == 0 {
			l.Errorw("No fromAddresses specified in VRF v2 job spec, please specify")
			continue
		}
		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddresses[0])
```

I think `GetRoundRobinAddress` was taking care of this for us before but now we need to do it ourselves here.

Reply:
should we enforce these checks in `validate.go` so we fail faster? 
1. check fromAddresses is non zero
2. check all addresses have same key-specific max gas prices

Reply:
For (1), it could be in `validate.go`, but (2) would have to be done in `delegate.go` since it has all the context (the `chain.Config()` object)

Reply:
Let me check if we don't do that already

Reply:
At the very least we should definitely check that fromAddresses has a non-zero length.